### ISSUE TITLE
Create CI pipeline to run mandatory SAP security scanners

### ISSUE THREAD

**Issue by tobiscr** [kyma-project/lifecycle-manager#48]
**Description**
The reconciler code base has to be scanned by the SAP security scanners. CI/CD-pipelines have to be created to run the mandatory scanner tools.

Mandatory scanners are:
* Protecode
* Checkmarx
* Whitesource

The code base in following repositories have to be scanned by these tools:
* Kyma Operator: https://github.com/kyma-project/kyma-operator
* Manifest Operator: https://github.com/kyma-project/manifest-operator
* Watcher: https://github.com/kyma-project/kyma-watcher

AC:
* Scanner tools are updated to support scanning the Kyma Operator repository and Manifest Operator repository (has to be aligned with @aberezovski)
* CI/CD pipelines are added which run the SAP security scanners (Whitesource, Protecode, Checkmarx) regularly for the Kyma Operator and Manifest Operator code bases. These pipelines are triggered at least once a day (Mo-Fr).

**Reason**
Be compliant with SAP product standards.

**Comment by jakobmoellersap** [comment]
Waiting for @TorstenD-SAP to setup an alignment with Pipeline Teams to setup Scan Pipelines

**Comment by jakobmoellersap** [comment]
Action Items:

- Adding Protecode Job and Whitesource Jobs in test-infra @jakobmoellersap (multiple go mods from watcher are resulting in individual scan behaviors)
- @aberezovski to setup step-by-step high level overview

**Comment by dekiel** [comment]
Data file for Whitesource periodic scans: https://github.com/kyma-project/test-infra/blob/main/templates/data/whitesource-periodics-data.yaml
Data file for Protecode presubmit scans: https://github.com/kyma-project/test-infra/blob/main/templates/data/protecode-guard-data.yaml
For daily Protecode scans please talk to @Halamix2. As far I remember this scans operate on some list of images to scan. There is an automation getting this list from values.yaml files but you can provide this list at your own too.

**Comment by aberezovski** [comment]
As discussed during the meeting I am going to describe the steps to be followed to onboard security scanning for new GitHub repositories:
- [ ] Checkmarx: @Halamix2 @aberezovski 
  - Create Checkmarx projects for each new GitHub Repository
  - Add new Checkmarx project ids to Kyma Security Scan Jenkins pipeline for Checkmarx
- [ ] Whitesource: @Halamix2 @jakobmoellersap 
  - Define a Prow job template/configuration to run Whitesource Unified Agent for each new GitHub repository similar like for [Reconciler](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/scans/whitesource-periodics.yaml#L345).
  >Important: in case multiple Go projects are defined in same GitHub repository it is better to scan each Go project separately that improve the transparency and accountability.
  Another recommendation is to add responsibility mapping for each new component (Go Project) per team name. The mapping is done in teams.yaml file in Github Tools.
- [ ] Protecode: @jakobmoellersap @Halamix2 
  - Gather the list of the images to be scanned and use it as input for Jenkins pipeline that performs container images scanning on daily basis.

**Comment by jakobmoellersap** [comment]
Whitesource Job definitions were kindly written already by @Halamix2 https://github.com/kyma-project/test-infra/pull/5795

**Comment by jakobmoellersap** [comment]
For Protecode I will first need to setup the image build jobs which we currently do not have

**Comment by Halamix2** [comment]
* Checkmarx scans were added, Jenkins pipeline was **not** updated
* Whitesource scan jobs were added

**Comment by aberezovski** [comment]
@Halamix2 @jakobmoellersap 
Until the components mentioned above are not part of the release, the Checkmarx Jenkins pipeline should skip those components to avoid misunderstand in context of minim security requirements required for upcoming release. Kyma security dashboard should not reflect more than it is required.
The PoC components SAST reported alerts could be audited direct in Checkmarx.

**Comment by jakobmoellersap** [comment]
The checkmarx audit is not included in the dashboard. For Whitesource I have heard already that it leads to side-effects. @Halamix2 @aberezovski is there a way to run the whitesource scans without including them in the dashboard?

**Comment by aberezovski** [comment]
@jakobmoellersap sure, that is possible.
I am not aware of any handbook/guide that will describe that. I guess it could be, at least I do not know about it.
The steps would be:
- Using [SVM](https://svmprod-zdohvhnx0v.dispatcher.int.sap.eu2.hana.ondemand.com/webapp/index.html#/WhitesourceSelfServiceEntry) create a test product in Whitesource. There is already a group for that purpose - [TEST - KYMA](https://svmprod-zdohvhnx0v.dispatcher.int.sap.eu2.hana.ondemand.com/webapp/index.html#/WhitesourceSelfServiceEntry/b20a335585c14b91af3c6c61bc013cbdaab23c150ce54dc197ea9d6e1577a3e9)
- Download [Whitesource Unified Agent (Java application)](https://docs.mend.io/bundle/unified_agent/page/getting_started_with_the_unified_agent.html)
- Use Whitesource [UA configuration for Go projects](https://github.com/kyma-project/test-infra/blob/main/prow/images/whitesource-scanner/go-mod-wss-unified-agent.config) 
- [here](https://github.com/kyma-project/test-infra/blob/main/prow/scripts/cluster-integration/helpers/start-wssagent.sh) you can see how it is scanned for Kyma

I lost all my scripts, due my MacBook died, so I could not just share them with you.
Let me know if you need support with that and I will try to prepare those scripts again.

**Comment by aberezovski** [comment]
@jakobmoellersap 
I found a [link](https://wiki.one.int.sap/wiki/display/osssec/Getting+Started+-+WhiteSource) from OS3 team that is responsible for all OSVM scanners.
Anyway, let me know if you need support on that.

**Comment by jakobmoellersap** [comment]
@Halamix2 I think this would be the best course of action to avoid confusing the release mgmt process.

**Comment by jakobmoellersap** [comment]
Prepared a draft for this change in https://github.com/kyma-project/test-infra/pull/5805 - I wasnt sure wether this is correct since I had to create a label for the Test Product (did not find any existing) so its in draft until now. Please confirm that everything is okay before I put it up for review @Halamix2

**Comment by jakobmoellersap** [comment]
Prepared a draft for this change in https://github.com/kyma-project/test-infra/pull/5805 - I wasnt sure wether this is correct since I had to create a label for the Test Product (did not find any existing) so its in draft until now. Please confirm that everything is okay before I put it up for review @Halamix2

**Comment by dekiel** [comment]
@jakobmoellersap please let @Halamix2 when you will be ready to move forward with this issue. For now I'm removing it from our project.

**Comment by jakobmoellersap** [comment]
@dekiel image builds are present but only on PRs, I am blocked by us not knowing how we do the Image Build on main going forward. Once this is solved I will ping here again and ask for the protecode jobs.

**Comment by jakobmoellersap** [comment]
@dekiel since we clarified the jobs and we now have post-submits running on all repositories the protecode image jobs are unblocked @Halamix2 . Is it clear how we set them up already or do we need a special flow here? AFAIK we only need to setup the images being present in the `resources` section of the control-plane. Ideally, I would like to have a job though where the scan happens on main of the actual repository though, that makes turnaround for sec-scanning much faster
